dashboard/app/models/script.rb

```diff
@@ -199,13 +199,16 @@ def self.stage_extras_script_ids
   # @param [User] user
   # @return [Script[]]
   def self.valid_scripts(user)
-    with_hidden = user.permission?(UserPermission::HIDDEN_SCRIPT_ACCESS)
+    user_experiments_enabled = user.any_experiments_enabled?
+    with_hidden = !user_experiments_enabled && user.hidden_script_access?
     cache_key = "valid_scripts/#{with_hidden ? 'all' : 'valid'}"
```

Comment:
this isn't quite right -- if the user has experiments enabled, we can't cache the response -- we need to compute the valid_scripts each time, so that users with different sets of experiments don't get the same response.

Reply:
ahh, sorry, I spoke too soon. it looks like you're doing the mapping after you fetch the cached result. this looks good.